ironfish-rust/src/frost_utils/split_spender_key.rs

```diff
@@ -0,0 +1,105 @@
+
+use group::GroupEncoding;
+use ironfish_frost::frost::{
+    frost::keys::IdentifierList,
+    keys::{KeyPackage, PublicKeyPackage},
+    Identifier,
+};
+use ironfish_zkp::{constants::PROOF_GENERATION_KEY_GENERATOR, ProofGenerationKey};
+use jubjub::SubgroupPoint;
+use rand::thread_rng;
+use std::collections::HashMap;
+
+use crate::{IncomingViewKey, OutgoingViewKey, PublicAddress, SaplingKey, ViewKey};
+
+use super::split_secret::{split_secret, SecretShareConfig};
+
+pub struct SpenderKeyConfig {
+    pub ak: [u8; 32],
+    pub pgk: ProofGenerationKey,
+    pub vk: ViewKey,
+    pub ivk: IncomingViewKey,
+    pub ovk: OutgoingViewKey,
+    pub address: PublicAddress,
+    pub key_packages: HashMap<Identifier, KeyPackage>,
+    pub pubkeys: PublicKeyPackage,
+}
+
+pub fn split_spender_key(
+    coordinator_sapling_key: SaplingKey,
+    min_signers: u16,
+    max_signers: u16,
+    secret: Vec<u8>,
+) -> SpenderKeyConfig {
+    let secret_config = SecretShareConfig {
+        min_signers,
+        max_signers,
+        secret,
+    };
+
+    let mut rng = thread_rng();
+    let (key_packages, pubkeys) =
+        split_secret(&secret_config, IdentifierList::Default, &mut rng).unwrap();
+
+    let authorizing_key_bytes = pubkeys.verifying_key().serialize();
+    let authorizing_key = Option::from(SubgroupPoint::from_bytes(&authorizing_key_bytes))
+        .expect("should be able to deserialize the verifying key into a SubgroupPoint");
+
+    let proof_generation_key = ProofGenerationKey {
+        ak: authorizing_key,
+        nsk: coordinator_sapling_key.sapling_proof_generation_key().nsk,
+    };
+
+    let nullifier_deriving_key = *PROOF_GENERATION_KEY_GENERATOR
+        * coordinator_sapling_key.sapling_proof_generation_key().nsk;
+
+    let view_key = ViewKey {
+        authorizing_key,
+        nullifier_deriving_key,
+    };
+
+    let incoming_viewing_key = IncomingViewKey {
+        view_key: SaplingKey::hash_viewing_key(&authorizing_key, &nullifier_deriving_key).unwrap(),
+    };
+
+    let outgoing_view_key: OutgoingViewKey = coordinator_sapling_key.outgoing_view_key().clone();
+
+    let public_address = incoming_viewing_key.public_address();
+    // SpenderKeyConfig {
+    //     authorizing_key.to_bytes(),
+    //     proof_generation_key,
+    //     view_key,
+    //     incoming_viewing_key,
+    //     outgoing_view_key,
+    //     public_address,
+    //     key_packages,
+    //     pubkeys,
+    // }
+
+    SpenderKeyConfig {
+        ak: authorizing_key.to_bytes(),
+        pgk: proof_generation_key,
+        vk: view_key,
+        ivk: incoming_viewing_key,
+        ovk: outgoing_view_key,
+        address: public_address,
+        key_packages,
+        pubkeys,
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_split_spender_key() {
+        let key = SaplingKey::generate_key();
+        let secret = key.spend_authorizing_key.to_bytes().to_vec();
+
```

Comment:
You could just do some basic assertions here for ak, vk, ivk, against the input sapling key. Then you can also check the public_address from the pubkeys output and verify it matches the input sapling key address, I think that test should be sufficient.